docs/build/message-history.md

```diff
@@ -0,0 +1,139 @@
+    env: 'production' as const,
+    enableV3: true,
+    appVersion: 'YourApp/1.0.0',
+    dbEncryptionKey: encryptionKey, // Replace with actual Array encryption key
+    // Add the history sync URL option
+    historySyncUrl: 'SYNC_URL'
+  }
+```
+
+</TabItem>
+<TabItem value="kotlin" label="Kotlin" attributes={{className: "kotlin_tab"}}>
+
+```kotlin
+Client().create(
+    account = alixWallet,
+    options = ClientOptions(
+        ClientOptions.Api(XMTPEnvironment.PRODUCTION, true),
+        enableV3 = true,
+        dbEncryptionKey: encryptionKey, // Replace with actual ByteArray encryption key
+        appContext = context
+        historySyncUrl = "SYNC_URL"
+    )
+)
+```
+
+</TabItem>
+<TabItem value="swift" label="Swift"  attributes={{className: "swift_tab"}}>
+
+```swift
+let options = ClientOptions(
+    api: ClientOptions.Api(
+        env: .production,
+        isSecure: true
+    ),
+    enableV3: true,
+    dbEncryptionKey: encryptionKey, // Replace with actual String encryption key
+    historySyncUrl: "SYNC_URL"
+)
+```
+
+</TabItem>
+<TabItem value="node" label="Node"  attributes={{className: "node_tab"}}>
+
+```tsx
+// Ry, this might be totallly off... Also, does Node support dbEncryptionKey: "encryptionKey"?
+
+// this API is experimental and may change in the future
+const client = await Client.create('0x1234...', {
+  apiUrl: 'SYNC_URL'
+});
+```
+
+</TabItem>
+</Tabs>
+
+## Request message history sync
+
+When a new client is created, it will look for other clients registered with the same inbox ID. 
+
+If there are none, there is no message history to be synced.
+
+If there are clients registered with the same inbox ID, it will request a message history sync with the clients. 
+
+For example, let’s say that inbox ID 1/client B requests a message history sync with inbox ID 1/client A.
+
+The request encodes, encrypts, and sends inbox ID 1/client A’s group chat messages to the message history server indicated by inbox ID 1/client A’s `historySyncUrl`. 
+
+The request then downloads the encrypted messages from the message history server to inbox ID 1/client B’s local database. Inbox ID 1/client B can then decrypt, decode, and display the messages.
+
+The request performs this sync for every client registered with inbox ID 1.
+
+![Sequence diagram illustrating the request message history sync flow between Inbox 1/Client B, Inbox 1/Client A, and a message history server](img/req-message-hist-sync.png)
+
+:::caution
+Downloading message history may involve a large amount of data. To help users avoid potential mobile data charges, consider prompting users to initiate the download only when connected to Wi-Fi.
+:::
+
+<Tabs groupId="sdk-langs">
+<TabItem value="rn" label="React Native" attributes={{className: "rn_tab"}}>
+
+```jsx
+await client.requestMessageHistorySync()
+```
+
+</TabItem>
+<TabItem value="kotlin" label="Kotlin" attributes={{className: "kotlin_tab"}}>
+
+```kotlin
+client.requestMessageHistorySync()
```

Comment:
```suggestion
client.requestHistorySync()
```

Reply:
I actually think this is named correctly 

https://github.com/xmtp/xmtp-android/blob/main/library/src/main/java/org/xmtp/android/library/Client.kt#L637-L639